### PR TITLE
Run linter tests on travis (was: Run linter before commit)

### PR DESCRIPTION
**Pull request by rejas**
MM has some linting tasks to keep the codestyle clean. Unfortunately they are not enforced upon git commits so some PR have a wring codestyle.

~This PR adds automatic linting on the pre-commit git hook~
This PR runs the linting tasks on travis builds to report issues found.

### CODE REVIEW

**Comment by Legion2** [comment]
Why not check linting in the travis ci for pull requests. And fail the pipeline if the codestyle is not conform the rules.

**Comment by MichMich** [comment]
I agree with @Legion2 - This also prevents the need for adding a dependency.

**Comment by rejas** [comment]
Ok, will rework my PR accordingly. Although I personally like to have my code linted before people send it over, reduces the round trip time ;-)
 
Question: Since grunt is only used for linting, would it be an option to remove that as a dependency too and call the linting tools directly?

**Comment by MichMich** [comment]
The less dependencies, the better ... :)

**Comment by Legion2** [comment]
@rejas The grunt task config must be changed, to report problems instead of fixing them. Can you give me push access to this branch then I can update the Gruntfile.

**Comment by rejas** [comment]
LIke this @Legion2  ?

**Comment by Legion2** [comment]
yes but now the automatic fixing of linting errors don't work anymore. So we have to create two configs for eslint, one that fixes all errors and the other only reports them without fixing them.
[Gruntfile.js.diff.txt](https://github.com/MichMich/MagicMirror/files/4317425/Gruntfile.js.diff.txt)

**Comment by rejas** [comment]
Done @MichMich 
Replacing grunt would be done on another day, couldnt find a good replacement for the grunt-jsonlint so far.

File: .eslintrc.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
 		"curly": "error",
 		"camelcase": ["error", {"properties": "never"}],
         "no-multiple-empty-lines": ["error", { "max": 1, "maxEOF": 1 }],
+		"no-multi-spaces": "error",
```
Comment (Legion2) [inline .eslintrc.json]: looks like `jsonlint` does not enforce one indent style

Comment (rejas) [reply .eslintrc.json]: Something one should look at when getting rid of the grunt dependencies...

**Pull request merged**